packages/client/src/workflow-client.ts

```diff
@@ -880,6 +1058,38 @@ export class WorkflowClient extends BaseClient {
         }
         return temporal.api.history.v1.History.create({ events });
       },
+      async executeUpdate<Ret, Args extends any[]>(
+        def: UpdateDefinition<Ret, Args> | string,
+        options?: WithArgs<Args, WorkflowUpdateOptions>
+      ): Promise<Ret> {
+        const next = this.client._executeUpdateHandler.bind(this.client);
+        const fn = interceptors.length ? composeInterceptors(interceptors, 'update', next) : next;
+        const { args, ...opts } = options ?? {};
+        return (await fn({
```

Comment:
How does the interceptor know if it is supposed to return the result or the handle? I think the interceptor should always return a handle, but the handle may have the outcome already on it. Yes that means unfortunately in the interceptor input you're gonna have to indicate the lifecycle stage you expect to run.

Reply:
Interceptors and the private "_startUpdateHandler" must return "raw values" (eg. a custom `XxxOutput` interface is probably the safest from a future-proof perspective; in the past, we have returned either protobuf response object or just a single field, such as runId, but I'd prefer we don't continue in that direction);.

Reply:
:+1: Returning an interface w/ a single field of the raw start update response proto works for me